socialhome/content/models.py

```diff
@@ -114,10 +115,17 @@ def save(self, author=None, *args, **kwargs):
         if self.pk:
             # Old instance, bust the cache
             self.bust_cache()
-        elif author:
-            # New with author, set a GUID and author
-            self.guid = uuid4()
-            self.author = author
+        else:
+            if author:
+                # New with author, set a GUID and author
+                self.guid = uuid4()
+                self.author = author
+
+            if self.pinned:
+                max_order = Content.objects.values_list("order").distinct().aggregate(order=Max("order"))["order"]
```

Comment:
Way shorter and a lot more performant is:

    Content.objects.aggregate(Max("order"))["order__max"]

But, you're taking *all* content here. We only care about the content of current user. And only pinned content.

Ps pro-tip: use the `python manage.py shell_plus` command to locally play around with these kind of things before writing code.